### PR TITLE
test/docs: document and expand test coverage for connection priority and retries

### LINKED ISSUE #218: [REFACTOR] Add connection priority support for WiFi
**Describe the issue**
The `ConnectionOptions.autoconnect_priority` field exists but isn't well documented or tested.

### PR DESCRIPTION
**Pull request by Biqydu**
This PR updates the documentation and adds additional tests for `ConnectionOptions`, particularly for `autoconnect_priority` and `autoconnect_retries`.

Although the implementation was already in place, the documentation needed clarification to better match NetworkManager's specifications. The test suite has also been expanded to cover boundary values, providing better confidence that the mapping behaves as expected.

Closes #218

### CODE REVIEW

File: nmrs/src/api/builders/wifi.rs [hunk 1]
```diff
@@ -412,4 +412,84 @@ mod tests {
...
+            Some(&Value::from(0i32))
+        );
+    }
+
+    #[test]
+    fn connection_with_max_priority_values() {
+        let opts = ConnectionOptions {
+            autoconnect: true,
+            autoconnect_priority: Some(i32::MAX),
+            autoconnect_retries: Some(i32::MAX),
```
Comment (cachebag) [inline nmrs/src/api/builders/wifi.rs]: https://networkmanager.dev/docs/api/1.44.4/settings-connection.html

NetworkManager documents the valid range as `-999..999`. This test would lock in invalid settings as expected behavior. Use `999` and `-999` instead.

File: nmrs/src/api/models/config.rs
Comment (cachebag) [inline nmrs/src/api/models/config.rs]: This should be changed too to say that `None` uses NM’s global default.

File: nmrs/src/api/builders/wifi.rs [hunk 2]
```diff
@@ -412,4 +412,84 @@ mod tests {
...
+            connection_section.get("autoconnect-retries"),
+            Some(&Value::from(i32::MAX))
+        );
+    }
+
+    #[test]
+    fn connection_with_min_priority_and_retries() {
+        let opts = ConnectionOptions {
+            autoconnect: true,
+            autoconnect_priority: Some(i32::MIN),
```
Comment (cachebag) [inline nmrs/src/api/builders/wifi.rs]: NetworkManager documents valid values as `-1..2147483647`, where `-1 = default` and `0 = forever`.

Use `-1`, `0`, `1`, and maybe `i32::MAX`.

https://networkmanager.dev/docs/api/1.44.4/settings-connection.html

File: nmrs/src/api/builders/wifi.rs [hunk 3]
```diff
@@ -412,4 +412,84 @@ mod tests {
...
+            autoconnect_retries: Some(i32::MIN),
```
Comment (cachebag) [inline nmrs/src/api/builders/wifi.rs]: https://github.com/freedesktop-rs/nmrs/pull/491/changes#r3561220278

**Review by cachebag — CHANGES_REQUESTED** [review]
LGTM just a few notes

**Comment by Biqydu** [comment]
> LGTM just a few notes

Thanks for the links to the official specification and the thorough review! 
You're completely right - my boundary choices were bad.

Pushed the fixes!

**Review by cachebag — APPROVED** [review]
